### PR TITLE
feature/gcsfs-and-fsspec-file-store-functions

### PR DESCRIPTION
**Pull request by isaacna**
**Pull request recommendations:**
- [x] Name your pull request _your-development-type/short-description_. Ex: _feature/read-tiff-files_
- [x] Link to any relevant issue in the PR description. Ex: _Resolves [gh-12], adds tiff file format support_
 - Part of work leading up to #16, will be used in event gather pipeline during audio creation 
- [x] Provide context of changes.
 - Adding file store functions using `gcsfs` and `fsspec` instead of the old filestore class from cdptools
 - Most of the logic surrounding the upload operation remain similar (deleting local files, checking for existing one first) 
- [x] Provide relevant tests for your feature or bug fix.
- [ ] Provide or update documentation for any feature added by your pull request.

Thanks for contributing!

### CODE REVIEW

**Comment by isaacna** [comment]
I think the test is failing on windows because of NamedTemporaryFile not working on Windows: https://docs.python.org/3/library/tempfile.html#tempfile.NamedTemporaryFile

Lmk if there's a better way to test `remove_local_file`, but I can look into this later

File: cdp_backend/file_store/functions.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+)
+log = logging.getLogger(__name__)
+
+GCS_URI = "gs://{bucket}/{filename}"
+
+###############################################################################
+
+
+def initialize_gcs_file_system(credentials_file: str) -> GCSFileSystem:
+    return GCSFileSystem(token=credentials_file)
```
Comment (JacksonMaxfield) [inline cdp_backend/file_store/functions.py:25]: Unfortunately we may need to have the credentials file as a param for get file uri and upload file and such just like I changed for the db upload functions.

Will see.

File: cdp_backend/file_store/functions.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+
+def get_file_uri(fs: GCSFileSystem, bucket: str, filename: str) -> str:
```
Comment (JacksonMaxfield) [inline cdp_backend/file_store/functions.py:28]: Return type should be `Union[str, None]` I think?

File: cdp_backend/tests/file_store/test_functions.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+)
+def test_remove_local_file() -> None:
+    f = NamedTemporaryFile()
+    f.write(b"Hello world!")
+
+    assert os.path.isfile(f.name)
+
+    functions.remove_local_file(f.name)
+
+    assert not os.path.isfile(f.name)
```
Comment (JacksonMaxfield) [inline cdp_backend/tests/file_store/test_functions.py:107]: You should be able to replace this with the pytest `tmpdir` fixture: https://docs.pytest.org/en/stable/tmpdir.html#the-tmpdir-fixture

**Review by JacksonMaxfield — APPROVED** [review]
Looks good to me minus some minor comments.

Can we get docstrings for everything though?

File: cdp_backend/file_store/functions.py (hunk 2 above)
Comment (isaacna) [reply cdp_backend/file_store/functions.py]: Good catch, I just changed it to Optional[str]

File: cdp_backend/file_store/functions.py (hunk 1 above)
Comment (isaacna) [reply cdp_backend/file_store/functions.py]: Ah I see, I think we'll definitely have to do that if we do multiprocessing so I went ahead and made the change

**Pull request merged**